docs/Client.asciidoc

```diff
@@ -0,0 +1,320 @@
+few examples of its use with `openqa-cli` earlier in the documentation.
+
+Here we will start again from the very beginning. To give you a more complete
+overview of its capabilities. To get started all you need is an openQA instance
+with a few jobs and `curl`. Just replace `openqa.example.com` in the examples
+below with the hostname of your openQA instance.
+
+[source,sh]
+----
+curl http://openqa.example.com/api/v1/jobs/overview
+----
+
+That one-liner will show you the latest jobs from the overview in JSON format.
+You could also append various query parameters to filter the jobs further.
+
+[source,sh]
+----
+curl http://openqa.example.com/api/v1/jobs/overview?result=failed
+----
+
+But not all REST endpoints are public. Some also require authentication with an
+API key and secret. And that's usually where you reach the limits of `curl`,
+because this key and secret are used to generate an individual checksum for
+every request.
+
+But openQA also contains a dedicated client to help you with that. It is called
+`openqa-cli` and can usually be installed with an `openQA-client`\ package (the
+name will vary depending on your Linux distribution).
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com jobs/overview result=failed
+----
+
+Our example above is quickly translated. The `api` subcommand of `openqa-cli`
+allows you to perform arbitrary HTTP requests against the REST API. The path
+will automatically get the correct version prefix applied (such as `/api/v1`),
+and query parameters can be passed along as `key=value` pairs.
+
+== Help
+
+The `api` subcommand is not the only one available and more will be added over
+time. To get a complete list of all currently available subcommands you can use
+the `--help` option.
+
+[source,sh]
+----
+openqa-cli --help
+----
+
+And each subcommand also contains descriptions for all its available options, as
+well as many common usage examples.
+
+[source,sh]
+----
+openqa-cli api --help
+----
+
+== Authentication
+
+Not all REST endpoints are public, many will return a `403 Forbidden` error if
+you try to access them without proper credentials. The credentials (or API keys)
+are managed in the web UI, to which you will need admin access.
+
+Once you have acquired a valid key and secret you can store them in a config
+file or use them ad-hoc from the command line. There are two config files
+`openqa-cli` will try, the global `/etc/openqa/client.conf`, and your personal
+`~/.config/openqa/client.conf`. The format is the same for both.
+
+[source,ini]
+----
+[openqa.example.com]
+key = 1234567890ABCDEF
+secret = 1234567890ABCDEF
+----
+
+For ad-hoc use all `openqa-cli` subcommands use the `--apikey` and `--apisecret`
+options. Which will override whatever the config files may contain.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com --apikey 1234567890ABCDEF \
+    --apisecret 1234567890ABCDEF isos/1
+----
```

Comment:
Seems in this example, the `apikey` and `apisecret` are not needed.  Changing an example with `-X post` should be better?